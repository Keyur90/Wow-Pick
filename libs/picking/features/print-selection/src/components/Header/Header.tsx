import IconButton from '@material-ui/core/IconButton';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import { Header as SharedHeader } from '@rf2/ui';
import React from 'react';
import { useHistory } from 'react-router-dom';

interface HeaderPropTypes {
  showTitle: boolean;
}

const useStyles = makeStyles((theme) => ({
  backButton: {
    marginRight: theme.spacing(2),
  },
}));

const Header: React.FC<HeaderPropTypes> = ({ showTitle }) => {
  const history = useHistory();
  const classes = useStyles();

  return (
    <SharedHeader>
      {showTitle && (
        <>
          <IconButton
            edge="start"
            className={classes.backButton}
            color="inherit"
            aria-label="back"
            onClick={() => history.push('/')}
          >
            <ArrowBackIcon />
          </IconButton>
          <Typography variant="h1" component="h1">
            Printer Selection
          </Typography>
        </>
      )}
    </SharedHeader>
  );
};

export { Header };
