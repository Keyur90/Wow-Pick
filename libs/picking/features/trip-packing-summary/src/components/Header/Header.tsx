import React from 'react';
import Typography from '@material-ui/core/Typography';
import { Header as SharedHeader } from '@rf2/ui';

const Header: React.FC = () => (
  <SharedHeader>
    <Typography variant="h1" component="h1">
      Trip Summary
    </Typography>
  </SharedHeader>
);

export { Header };
