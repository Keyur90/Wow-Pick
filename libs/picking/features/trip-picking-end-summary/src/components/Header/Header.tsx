import React from 'react';
import Typography from '@material-ui/core/Typography';
import { Header as SharedHeader } from '@rf2/ui';

const Header: React.FC = () => (
  <SharedHeader>
    <Typography variant="h1" component="h1">
      Trip Status
    </Typography>
  </SharedHeader>
);

export { Header };
