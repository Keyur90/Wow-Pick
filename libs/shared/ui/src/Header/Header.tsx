import AppBar from '@material-ui/core/AppBar';
import { makeStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import React, { ReactNode } from 'react';

interface HeaderPropsType {
  children?: ReactNode;
}

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    position: 'relative',
    margin: theme.spacing(0, 0, 7),
  },
}));

const useAppBarStyles = makeStyles((theme) => ({
  root: {
    background: theme.palette.primary.dark,
    color: theme.palette.common.white,
  },
}));

const useToolbarStyles = makeStyles((theme) => ({
  regular: {
    minHeight: 56,
  },
}));

const Header: React.FC<HeaderPropsType> = ({ children }) => {
  const classes = useStyles();
  const appBarclasses = useAppBarStyles();
  const toolbarClasses = useToolbarStyles();

  return (
    <div className={classes.root}>
      <AppBar classes={appBarclasses} position="fixed">
        <Toolbar classes={toolbarClasses}>{children}</Toolbar>
      </AppBar>
    </div>
  );
};

export { Header };
