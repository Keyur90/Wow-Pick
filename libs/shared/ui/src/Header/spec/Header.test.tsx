import React from 'react';
import { render } from '@testing-library/react';
import { Header } from '../Header';

test('renders Header', async () => {
  const { container } = render(
    <Header>
      <h1>Trip summary</h1>
    </Header>
  );

  expect(container).toMatchSnapshot();
});
