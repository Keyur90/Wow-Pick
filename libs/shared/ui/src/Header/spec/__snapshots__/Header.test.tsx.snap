// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root makeStyles-root-2 MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-regular-3 MuiToolbar-gutters"
      >
        <h1>
          Trip summary
        </h1>
      </div>
    </header>
  </div>
</div>
`;
